src/types.ts

```diff
@@ -19,7 +19,7 @@ export interface UnpluginOptions {
   transformInclude?: (id: string) => boolean;
   transform?: (this: UnpluginContext, code: string, id: string) => Thenable<TransformResult>;
   load?: (this: UnpluginContext, id: string) => Thenable<TransformResult>
-  resolveId?: (id: string, importer?: string) => Thenable<string | null | undefined>
+  resolveId?: (id: string, importer?: string) => Thenable<string | null | undefined | object>
```

Comment:
We should have bea better type for it instead of arbitrary `object`. Could refer to Rollup's type definiation.

We will also need to change some implementation in webpack to handle the object style.